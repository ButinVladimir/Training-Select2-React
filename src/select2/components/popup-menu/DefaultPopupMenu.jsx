import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

export default function DefaultPopupMenu({
  inputTextField,
  itemsList,
  onBlur,
}) {
  return (
    <div className="select2-popup-menu" onBlur={onBlur}>
      <div className="input-text-container">
        {inputTextField}
      </div>
      <div>
        {itemsList}
      </div>
    </div>
  );
}

DefaultPopupMenu.propTypes = {
  inputTextField: PropTypes.element.isRequired,
  itemsList: PropTypes.element.isRequired,
  onBlur: PropTypes.func.isRequired,
};
