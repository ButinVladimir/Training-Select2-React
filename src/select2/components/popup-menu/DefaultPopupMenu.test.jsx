import React from 'react';
import renderer from 'react-test-renderer';
import DefaultPopupMenu from './DefaultPopupMenu';

describe('DefaultPopupMenu', () => {
  test('renders without crashing', () => {
    const component = renderer.create(<DefaultPopupMenu
      inputTextField={<span>Text field</span>}
      itemsList={<span>Items list</span>}
    />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
