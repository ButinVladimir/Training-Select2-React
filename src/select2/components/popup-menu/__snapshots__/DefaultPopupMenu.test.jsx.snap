// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultPopupMenu renders without crashing 1`] = `
<div
  className="select2-popup-menu"
>
  <div
    className="input-text-container"
  >
    <span>
      Text field
    </span>
  </div>
  <div>
    <span>
      Items list
    </span>
  </div>
</div>
`;
